### PR TITLE
PICARD-357: Fix the way Picard handles filenames with only extensions

### PR DESCRIPTION
**Pull request by samj1912**
https://tickets.metabrainz.org/browse/PICARD-357

### CODE REVIEW

File: picard/file.py [hunk 1]
```diff
@@ -284,6 +284,14 @@ def _make_filename(self, filename, metadata, settings=config.setting):
 
         if settings["rename_files"]:
             new_filename, ext = os.path.splitext(new_filename)
+            try:
+                # In case the filename is blank and only has the extension
+                # the real extension is in new_filename and ext is blank
+                if ext == '' and new_filename.lower() in map(unicode, self.EXTENSIONS):
+                    ext = new_filename
+                    new_filename = ''
+            except AttributeError:
```
Comment (mineo) [inline picard/file.py]: Under which circumstances can the lines above raise an AttributeError?

Comment (samj1912) [inline picard/file.py]: In case extensions is not defined

Comment (mineo) [inline picard/file.py]: `register_formats` and `supported_formats` already don't work if `EXTENSIONS` is not a defined attribute.

**Comment by samj1912** [comment]
Updated :)

**Comment by zas** [comment]
@samj1912 : can you update tests so this case is tested ?

**Comment by samj1912** [comment]
sure :)

**Pull request merged**